tests/test_strategy_infinite_bus.py

```diff
@@ -306,30 +306,88 @@ def testing_event_market_cycle_post_offers(bus_test3, area_test3):
 
 
 @pytest.fixture()
-def area_test4():
-    return FakeArea(0)
-
-
-@pytest.fixture()
-def bus_test4(area_test4):
+def bus_test4(area_test1):
     c = InfiniteBusStrategy(energy_sell_rate=30, energy_buy_rate=25)
-    c.area = area_test4
-    c.owner = area_test4
+    c.area = area_test1
+    c.owner = area_test1
     return c
 
 
-def testing_event_tick_buy_energy(bus_test4, area_test4):
+def testing_event_tick_buy_energy(bus_test4, area_test1):
     bus_test4.event_activate()
     bus_test4.event_tick()
-    assert len(area_test4.test_market.traded_offers) == 1
-    assert area_test4.test_market.traded_offers[-1].offer.energy == 1
+    assert len(area_test1.test_market.traded_offers) == 1
+    assert area_test1.test_market.traded_offers[-1].offer.energy == 1
 
 
-def testing_event_market_cycle_posting_bids(bus_test4, area_test4):
+def testing_event_market_cycle_posting_bids(bus_test4, area_test1):
     ConstSettings.IAASettings.MARKET_TYPE = 2
     bus_test4.event_activate()
     bus_test4.event_market_cycle()
     assert len(bus_test4._bids) == 1
-    assert bus_test4._bids[area_test4.test_market.id][-1].energy == sys.maxsize
-    assert bus_test4._bids[area_test4.test_market.id][-1].price == 25 * sys.maxsize
+    assert bus_test4._bids[area_test1.test_market.id][-1].energy == sys.maxsize
+    assert bus_test4._bids[area_test1.test_market.id][-1].price == 25 * sys.maxsize
     ConstSettings.IAASettings.MARKET_TYPE = 1
+
+
+def test_global_market_maker_rate_single_value(bus_test4):
+    assert isinstance(GlobalConfig.market_maker_rate, int)
+    assert (GlobalConfig.market_maker_rate ==
+            ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE)
+
+
+"""TEST5"""
+
+
+@pytest.fixture()
+def bus_test5(area_test1):
+    c = InfiniteBusStrategy(energy_rate_profile="src/d3a/resources/SAM_SF_Summer.csv")
+    c.area = area_test1
+    c.owner = area_test1
+    return c
+
+
+def test_global_market_maker_rate_profile(bus_test5):
+    assert isinstance(GlobalConfig.market_maker_rate, dict)
+    assert len(GlobalConfig.market_maker_rate) == 96
+    assert list(GlobalConfig.market_maker_rate.values())[0] == 516.0
+    assert list(GlobalConfig.market_maker_rate.values())[-1] == 595.0
+
+
+"""TEST6"""
+
+
+@pytest.fixture()
+def bus_test6(area_test1):
+    c = InfiniteBusStrategy(energy_rate_profile="src/d3a/resources/SAM_SF_Summer.csv")
+    c.area = area_test1
+    c.owner = area_test1
+    return c
+
+
+def test_infinite_bus_selling_rate_set_as_profile(bus_test6):
+    bus_test6.event_activate()
+    assert isinstance(bus_test6.energy_rate, dict)
+    assert len(bus_test6.energy_rate) == 96
+    assert list(bus_test6.energy_rate.values())[0] == 516.0
+    assert list(bus_test6.energy_rate.values())[-1] == 595.0
+
```

Comment:
IMO you could combine these methods. If you want to separate the check for the `energy_rate` and the `market_maker_rate`, ok, but you can definitely reuse the `bus_test5` fixture in `test_infinite_bus_selling_rate_set_as_profile `

Reply:
Okay, will do so BUT please try to catch/drop such comments in earlier rounds of review, so could be adapted within lesser number review rounds

Reply:
Yes, sorry, my bad. I thought I'd look into the tests once the overall code is ready. I will do it right away next time.

Reply:
But I have to say: https://github.com/gridsingularity/d3a/pull/1167#discussion_r665177470 Is basically the same comment and you could have adapted ti the bus fixtures too.

Reply:
Not really AFAIS, https://github.com/gridsingularity/d3a/pull/1167#discussion_r665177470 was to reuse the same fixtures whereas this thread is to merge two test into one targeting different assertions